### ISSUE TITLE
Android buildbot builder fails

### ISSUE THREAD

**Issue by Ms2ger** [servo/servo#14571]
Followup from #13154.

```
DEBUG: configure: error: Building ICU requires either yasm or a GNU assembler. If you do not have either of those available for this platform you must use --without-intl-api
ERROR: old-configure failed
make: *** [maybe-configure] Error 1
thread 'main' panicked at 'assertion failed: result.success()', /home/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f7917c4/build.rs:43
stack backtrace:
   1:     0x7fa793586a4a - std::sys::imp::backtrace::tracing::imp::write::h944c02ac40aee2d7
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:42
   2:     0x7fa79358a81f - std::panicking::default_hook::{{closure}}::h6875a2976258b020
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:247
   3:     0x7fa79358a41e - std::panicking::default_hook::h88ffbc5922643264
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:263
   4:     0x7fa79358ac07 - std::panicking::rust_panic_with_hook::ha5aed1dfc0e220e3
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:451
   5:     0x7fa7935748b3 - std::panicking::begin_panic::h5f0a1fd8f4de57ec
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:413
   6:     0x7fa79357e5fd - build_script_build::main::h7a38f73d1d623d20
                        at /home/servo/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/f7917c4/build.rs:43
   7:     0x7fa79359550a - __rust_maybe_catch_panic
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libpanic_unwind/lib.rs:97
   8:     0x7fa79358b20a - std::rt::lang_start::h5d71a3afaaa4b2ff
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panicking.rs:332
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/panic.rs:351
                        at /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/obj/../src/libstd/rt.rs:57
   9:     0x7fa79357ec62 - main
  10:     0x7fa79296af44 - __libc_start_main
  11:     0x7fa793571488 - <unknown>
  12:                0x0 - <unknown>

[Warning] Could not generate notification! Optional Python module 'dbus' is not installed.
Build FAILED in 0:07:03
program finished with exit code 101
```

**Comment by larsbergstrom** [comment]
I believe that this is due to the presence of the `ANDROID_TOOLCHAIN` variable, which has been removed but still is sticking around due to https://github.com/servo/saltfs/issues/552

**Comment by larsbergstrom** [comment]
The error is now:
```
 (view as text)

python ./etc/ci/check_dynamic_symbols.py
 in dir /home/servo/buildbot/slave/android/build (timeout 1200 secs)
 watching logfiles {}
 argv: ['python', './etc/ci/check_dynamic_symbols.py']
 environment:
  ANDROID_NDK=/home/servo/android/ndk/current/
  ANDROID_SDK=/home/servo/android/sdk/current/
  CARGO_HOME=/home/servo/.cargo
  CCACHE=/usr/bin/ccache
  DISPLAY=:0
  HOME=/home/servo
  PATH=/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin:/home/servo/android/sdk/current/platform-tools
  PWD=/home/servo/buildbot/slave/android/build
  RUST_BACKTRACE=1
  SERVO_CACHE_DIR=/home/servo/.servo
  SHELL=/bin/bash
  TERM=linux
  UPSTART_INSTANCE=
  UPSTART_JOB=buildbot-slave
 using PTY: False
Unexpected dynamic symbols in binary:
cargo_apk_injected_glue_write_log
cargo_apk_injected_glue_get_native_window
__cxa_type_match
cargo_apk_injected_glue_add_sender
cargo_apk_injected_glue_set_multitouch
android_main
program finished with exit code 255
elapsedTime=0.094732
```

@mmatyas Should we add all of those symbols to the list of permitted ones?

**Comment by larsbergstrom** [comment]
I suspect that once the injected glue PR lands (#14530), the only one we will have to add to the list is `__cxa_type_match`.

**Comment by mmatyas** [comment]
Yes, that's right, you'll only need `__cxa_type_match`, the rest of them are fixed by #14530.

**Comment by Ms2ger** [comment]
Confirmed that the latest build only has
```
Unexpected dynamic symbols in binary:
__cxa_type_match
```
